onnxruntime/python/tools/quantization/neural_compressor/weight_only.py

```diff
@@ -0,0 +1,1003 @@
+#
+#  The implementation of this file is based on:
+# https://github.com/intel/neural-compressor/tree/master/neural_compressor
+#
+# Copyright (c) 2023 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Modifications:
+# Add k-quant quantization method.
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""WeightOnly for onnxrt adaptor."""
+
+import copy
+import logging
+import os
+import struct
+import sys
+
+import numpy as np
+import onnx
+from onnx import numpy_helper
+from packaging.version import Version
+
+import onnxruntime as ort
+
+from .onnx_model import ONNXModel
+from .util import dtype_mapping, simple_progress_bar
+
+logger = logging.getLogger("neural_compressor")
+ONNXRT116_VERSION = Version("1.16.0")
+ONNXRT1161_VERSION = Version("1.16.1")
+ONNXRT1211_VERSION = Version("1.21.1")
+
+
+def get_blob_size(group_size, num_bits, has_zp):  # pragma: no cover
+    """Get blob_size.
+
+    Args:
+        group_size (int): how many elements share one scale/zp
+        has_zp (bool): whether zero_point is None
+    """
+    if Version(ort.__version__) > ONNXRT1161_VERSION:
```

Comment:
Do not need handle older version of ORT since in main branch, which means current ORT version is latest.